save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt

```diff
@@ -210,6 +206,42 @@ class DownloadFilesController(
         }
     }
 
+    private fun getExecutionId(testExecutionDto: TestExecutionDto): Long {
+        testExecutionDto.executionId?.let { return it }
+
+        val agentContainerId = testExecutionDto.agentContainerId
+            ?: throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Request body should contain agentContainerId")
+        val execution = agentRepository.findByContainerId(agentContainerId)?.execution
+        return execution?.id
+            ?: throw ResponseStatusException(
+                HttpStatus.NOT_FOUND,
+                "Execution for agent $agentContainerId not found"
+            )
+    }
+
+    /**
+     * @param testExecutionDto
+     * @return [Mono] with response
```

Comment:
this method returns `String`, not `Mono<String>`

Reply:
It is wrapped to mono (Mono.just(result)) by webflux in InvocableHandlerMethod.invoke

Reply:
interesting, but here we return String. KDoc describes this method